ophyd/device.py

```diff
@@ -207,22 +257,33 @@ def __repr__(self):
         suffix = (repr(self.suffix) if self.suffix
                   else '')
 
-        component_class = self.cls.__name__
+        component_class = self.cls.__name__ if self.cls else ""
         args = ', '.join(s for s in (component_class, suffix, kw_str) if s)
 
         this_class = self.__class__.__name__
         return f'{this_class}({args})'
 
     __str__ = __repr__
 
-    def __get__(self, instance, owner):
+    @typing.overload
+    def __get__(self, instance: None, owner: type) -> Component:
+        ...
+
+    @typing.overload
+    def __get__(self, instance: Any, owner: type) -> K:
+        ...
```

Comment:
Is this the core bit that makes the static typing analysis work? I feel like I need to read up more on the internals of how type hinting generics work to give a comprehensive review. I'm going to poke around with with branch locally to observe the effects.

Reply:
This `__get__` block is what allows the [descriptor](https://docs.python.org/3/howto/descriptor.html) mechanism utilized by `Component` to describe what it does to the static analyzer: 
* Overloading indicates the two scenarios (1) access by class where you get back a `Component` instance itself, and (2) access by instance where you get back the instantiated `cpt.cls`.
* Both the generic `Component[K]` and this section are required for the instance `__get__` to work; otherwise the typevar `K` couldn't be used here.